File: Cab Booking Frontend/cabBooking/src/app/services/car.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Car } from '../model/car';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CarService {

  private baseUrl = 'http://localhost:8081/api/cars';

  constructor(private http : HttpClient) { }

  addCar(car: Car): Observable<Car> {
    return this.http.post<Car>(`${this.baseUrl}/addCar`, car);
  }

  findAll(): Observable<Car[]> {
    return this.http.get<Car[]>(`${this.baseUrl}`);
  }

  getCarById(id: number): Observable<Car> {
    return this.http.get<Car>(`${this.baseUrl}/${id}`);
  }

  getCarsByDriverId(driverId: number): Observable<Car[]> {
    return this.http.get<Car[]>(`${this.baseUrl}/driver/${driverId}`);
  }

  getCarsByCategory(category: string): Observable<Car[]> {
    return this.http.get<Car[]>(`${this.baseUrl}/category/${category}`);
  }

  deleteById(carNo: number): Observable<boolean> {
    return this.http.delete<boolean>(`${this.baseUrl}/${carNo}`);
  }

  updateCar(carNo: number, carDetails: Car): Observable<Car> {
    return this.http.put<Car>(`${this.baseUrl}/${carNo}`, carDetails);
  }
  
}

File: Cab Booking Backend/CarService/src/main/java/com/carService/service/CarService.java
package com.carService.service;

import java.util.List;

import com.carService.entity.Car;
import com.carService.entity.CarCategory;

public interface CarService {
	Car addCar(Car car);
	Car updateCar(Long id, Car carDetails);
    void deleteCar(Long id);
    Car getCarById(Long id);
    List<Car> getAllCars();
    List<Car> getAvailableCars();
    List<Car> getCarsByCategory(CarCategory category);
    List<Car> getCarsByDriverId(Long driverId);
}
